apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/migrate-to-core-fse.php

```diff
@@ -0,0 +1,142 @@
+		return $errors;
+	}
+
+	$dotcom_fse_theme = get_stylesheet();
+	// Core FSE themes that are forks of Dotcom FSE equivalents will have "-blocks" appended to their name.
+	$core_fse_theme = sprintf( '%s-blocks', $dotcom_fse_theme );
+
+	// We can't migrate if the fork theme that supports core FSE is not present.
+	$new_theme = wp_get_theme( $core_fse_theme );
+	if ( ! $new_theme->exists() || ! $new_theme->is_block_theme() ) {
+		$message = sprintf( "We were looking for the %s theme and it doesn't seem to exist", $core_fse_theme );
+		$errors->add( 403, $message );
+		return $errors;
+	}
+
+	// Activate theme fork that supports Core FSE. This will also deactivate Dotcom FSE.
+	switch_theme( $core_fse_theme );
+	// Delete the option that `did_insert_templates()` created to ensure we won't reactivate.
+	delete_option( basename( $dotcom_fse_theme ) . ' -fse-template-data-v1' );
+	$notes[] = sprintf( 'Now running the %s theme', $core_fse_theme );
+
+	$posts = get_posts( array( 'post_type' => 'wp_template_part' ) );
+
+	$count = 0;
+	foreach ( $posts as $post ) {
+		// Once Dotcom FSE is disabled a8c blocks will no longer be availble.
+		// They are replaced with the maintained Core blocks.
+		$data = array(
+			'ID'           => $post->ID,
+			'post_content' => wpcom_replace_a8c_fse_blocks( $post->post_content ),
+		);
+
+		wp_update_post( $data );
+		wpcom_add_core_template_part_taxonomies( $post );
+		$count++;
+	}
+	$notes[] = sprintf( '%d template parts migrated', $count );
+
+	\A8C\FSE\activate_core_fse();
+	return $notes;
+}
+
+/**
+ * Replaces Legacy FSE blocks with their Core doppelgangers.
+ *
+ * @param string $post_content Content to be transformed.
+ * @return string Replaced post_content
+ */
+function wpcom_replace_a8c_fse_blocks( $post_content ) {
+	$conversion_map    = array(
+		'a8c/site-title'       => 'core/site-title',
+		'a8c/site-description' => 'core/site-tagline',
+		'a8c/navigation-menu'  => 'core/navigation',
+	);
+	$blocks_to_convert = array_keys( $conversion_map );
+	$parsed_blocks     = parse_blocks( $post_content );
+	foreach ( $parsed_blocks as $i => $block ) {
+		if ( ! in_array( $block['blockName'], $blocks_to_convert, true ) ) {
+			continue;
+		}
+		$block['blockName'] = $conversion_map[ $block['blockName'] ];
+
+		// navigation needs a helping hand.
+		if ( 'core/navigation' === $block['blockName'] ) {
+			$block['attrs']['__unstableLocation'] = 'primary';
+			// a8c nav's default alignment is center in the absence of alignment.
+			$align                    = $block['attrs']['textAlign'] ?? 'center';
+			$block['attrs']['layout'] = array(
+				'type'           => 'flex',
+				'justifyContent' => $align,
+				'orientation'    => 'horizontal',
+			);
+		}
+
+		// site title and tagline need to be centered if they aren't.
+		if ( in_array( $block['blockName'], array( 'core/site-title', 'core/site-tagline' ), true ) ) {
+			$block['attrs']['textAlign'] = $block['attrs']['textAlign'] ?? 'center';
+		}
```

Comment:
Note that these alignment settings may be theme dependent. So these make sense for Maywood for now, but for other themes may need to be handled differently.